extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -980,20 +966,43 @@ public String getHost() {
 
     @Nonnull
     @Override
-    public String getPrivacy() {
-        return "";
+    public String getPrivacy() throws ParsingException {
+        boolean isUnlisted = playerResponse
+                .getObject("microformat")
+                .getObject("playerMicroformatRenderer")
+                .getBoolean("isUnlisted");
+        return isUnlisted ? "Unlisted" : "Public";
```

Comment:
Do we have the possible options documented somewhere? And shouldn't we maybe use e.g. an `Enum` instead of just returning a `String`? Also for the `getLicence()` and maybe even for the `getCategory()` as well.

Reply:
Yes it is documented: https://support.google.com/youtube/answer/157177 "Click the down arrow under "Visibility" and choose Public, Private, or Unlisted".
I'll test again with private videos, but in my tests long ago iirc it was ok (it show only content not available I think).

<sub><sup>I'll see later for Enums</sup></sub>

Reply:
I meant in NewPipe. And what if another service calls e.g. "Public" "World Visible" or whatever? And you have to consider translations in NewPipe as well.

Reply:
Yes, I've already created [some translators](https://github.com/B0pol/NewPipe/tree/metadata/app/src/main/java/org/schabi/newpipe/util/translators) and have choosen Private / Public / Unlisted

Note to self: todo: add internal privacy for PeerTube.

Reply:
Yes, an enum would be better for privacy. For licence, it's not possible because it changes dynamically based on companies owning the copyrights.

Reply:
@B0pol: I rebased, removed Eclipse crap, and added a `Privacy` enum.